Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidNoWarnAnalyzerSuppressionAnalyzer.cs

```diff
@@ -0,0 +1,163 @@
+// © 2025 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class AvoidNoWarnAnalyzerSuppressionAnalyzer : SolutionAnalyzer
+	{
+		private const string Title = @"Avoid NoWarn for analyzer suppression";
+		private const string MessageFormat = @"Use .editorconfig instead of NoWarn project setting to suppress analyzer '{0}'. Configure with 'dotnet_diagnostic.{0}.severity = none'.";
+		private const string Description = @"NoWarn project settings should be avoided for analyzer suppression. Use .editorconfig files for better maintainability and team consistency.";
+
+		private static readonly char[] SeparatorChars = { ';', ',' };
+
+		public AvoidNoWarnAnalyzerSuppressionAnalyzer()
+			: base(DiagnosticId.AvoidNoWarnAnalyzerSuppression, Title, MessageFormat, Description, Categories.Maintainability, isEnabled: false)
+		{
+		}
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.EnableConcurrentExecution();
+
+			context.RegisterCompilationAction(AnalyzeCompilation);
+		}
+
+		private void AnalyzeCompilation(CompilationAnalysisContext context)
+		{
+			try
+			{
+				var projectFilePath = TryFindProjectFileFromSourcePaths(context);
+				if (string.IsNullOrEmpty(projectFilePath) || !File.Exists(projectFilePath))
+				{
+					return;
+				}
+
+				AnalyzeProjectFile(context, projectFilePath);
+			}
+			catch (Exception)
+			{
+				// Silently ignore errors to avoid breaking compilation
+				return;
+			}
+		}
+
+		private void AnalyzeProjectFile(CompilationAnalysisContext context, string projectFilePath)
+		{
+			try
+			{
+				var content = File.ReadAllText(projectFilePath);
+				var document = XDocument.Parse(content);
+
+				IEnumerable<XElement> noWarnElements = document.Descendants()
+					.Where(e => e.Name.LocalName.Equals("NoWarn", StringComparison.OrdinalIgnoreCase));
+
+				foreach (XElement noWarnElement in noWarnElements)
+				{
+					var noWarnValue = noWarnElement.Value;
+					if (string.IsNullOrWhiteSpace(noWarnValue))
+					{
+						continue;
+					}
+
+					// Find PH analyzer codes in NoWarn value
+					List<string> phCodes = ExtractAnalyzerCodes(noWarnValue);
```

Comment:
I think it's reasonable to complain about any usage of NoWarn. For niche use cases (3rd party stuff, legacy stuff, etc., it would make sense to just disable this rule.)

If we go this route, we presumably could eliminate the overhead of XDocument as well. Just search for the existence of <NoWarn>?